lib/prometheus/client/data_stores/direct_file_store.rb

```diff
@@ -292,22 +292,18 @@ def init_value(key)
             @positions[key] = @used - 8
           end
 
-          # Yield (key, value, pos). No locking is performed.
+          # Read position of all keys. No locking is performed.
           def read_all_values
             @f.seek(8)
-            values = []
             while @f.pos < @used
               padded_len = @f.read(4).unpack('l')[0]
-              encoded = @f.read(padded_len).unpack("A#{padded_len}")[0]
-              value = @f.read(8).unpack('d')[0]
-              values << [encoded.strip, value, @f.pos - 8]
+              key = @f.read(padded_len).unpack("A#{padded_len}")[0].strip
+              @positions[key] = @f.pos
+              @f.seek(8, :CUR)
```

Comment:
Ooooh, this is so much better
Can we name this method `populate_positions`, or something like that?

Other than that, :+1: from me
I'd check with @Sinjo, though, it's pretty much his repo now :D

Btw, i'm assuming you've updated this repo to the latest master on the main repo, now that #95 is merged?
If not, probably a good idea to do that.

Reply:
> Btw, i'm assuming you've updated this repo to the latest master on the main repo, now that prometheus#95 is merged? If not, probably a good idea to do that.

Nope, this PR is based on `gc_production_branch_do_not_push` of this forked repo (which in turn is just a copy of https://github.com/prometheus/client_ruby/pull/95 I think?). Probably better to catchup with @Sinjo to understand if we want to get these changes tested in production.